src/FRESteamWorks/FREConverters.cpp

```diff
@@ -65,6 +65,61 @@ FREObject FREArray(uint32 length) {
 	return array;
 }
 
+FREObject FREBitmapDataFromImageRGBA(uint32 width, uint32 height, uint8* pImageRGBA)
+{
+	FREObject freWidth;
+	FRENewObjectFromUint32(uint32_t(width), &freWidth);
+
+	FREObject freHeight;
+	FRENewObjectFromUint32(uint32_t(height), &freHeight);
+
+	FREObject freTransparent;
+	FRENewObjectFromBool(uint32_t(0), &freTransparent);
+
+	FREObject freFillColor;
+	FRENewObjectFromUint32(uint32_t(0xFF0000), &freFillColor);
+
+	FREObject freArguments[4] = { freWidth, freHeight, freTransparent, freFillColor };
+
+	FREObject freoBitmapData;
+	FRENewObject((uint8_t *)"flash.display.BitmapData", 4, freArguments, &freoBitmapData, NULL);
+
+	FREBitmapData freBitmapData;
+	FREAcquireBitmapData(freoBitmapData, &freBitmapData);
+
+	int x, y;
+	// There may be extra pixels in each row due to the value of
+	// bmd.lineStride32, we'll skip over those as needed
+	int offset = freBitmapData.lineStride32 - freBitmapData.width;
+	uint32_t *bmdPixels = freBitmapData.bits32;
+
+	int index = 0;
+	int bitmapWidth = freBitmapData.width;
+	int bitmapHeight = freBitmapData.height;
+	for (y = bitmapHeight - 1; y > -1; y--)
```

Comment:
It is very strange, because in my case bitmap returns flipped when i use (for (y = 0; y < bitmapHeight; y++)) loop. Maybe it depends of steam sdk version?

Reply:
Mh, that's interesting. I was using Steam SDK v141 when testing this. Maybe it also depends on the AIR SDK or platform? If you don't want to debug this, just leave it as is and I'll figure out what's going on.

Reply:
Fixed this problem. Used FREBitmapData2. It contains one additional field, isInvertedY, which indicates the order in which the rows of image data are stored. [link](http://help.adobe.com/en_US/air/extensions/WS460ee381960520ad-866f9c112aa6e1ad46-7ff3.html)